lib/phoenix/live_dashboard/components/table_component.ex

```diff
@@ -0,0 +1,282 @@
+  #   `(params(), node()) -> {list(), integer() | binary()}`
+  # * `:rows_name` - A string to name the representation of the rows.
+  #   Default is calculated with the given `:page_name`.
+  # * `:title` - The title of the table.
+  #   Default is calculated with the given `:page_name`.
+
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @sort_dir ~w(desc asc)
+  @limit [50, 100, 500, 1000, 5000]
+
+  @type params() :: %{
+          limit: pos_integer(),
+          sort_by: :atom,
+          sort_dir: :desc | :asc,
+          search: binary()
+        }
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{
+      columns: columns,
+      node: node,
+      page_name: page_name,
+      params: all_params,
+      row_fetcher: row_fetcher
+    } = assigns
+
+    limit_options = assigns[:limit_options] || @limit
+    columns = normalize_columns(columns)
+    table_params = normalize_table_params(all_params, columns, limit_options)
+    {rows, total} = row_fetcher.(table_params, node)
+
+    {:ok,
+     assign(socket,
+       all_params: all_params,
+       columns: columns,
+       limit_options: limit_options,
+       node: node,
+       page_name: page_name,
+       row_attrs: assigns[:row_attrs] || [],
+       row_fetcher: row_fetcher,
+       rows: rows,
+       rows_name: assigns[:rows_name] || Phoenix.Naming.humanize(page_name) |> String.downcase(),
+       table_params: table_params,
+       title: assigns[:title] || Phoenix.Naming.humanize(page_name),
+       total: total
+     )}
+  end
+
+  defp normalize_columns(columns) do
+    Enum.map(columns, fn %{field: field} = column ->
+      column
+      |> Map.put_new_lazy(:header, fn -> Phoenix.Naming.humanize(field) end)
+      |> Map.put_new(:header_attrs, [])
+      |> Map.put_new(:format, & &1[field])
+      |> Map.put_new(:cell_attrs, [])
+      |> Map.put_new(:sortable, false)
+    end)
+  end
+
+  defp normalize_table_params(all_params, columns, limit_options) do
+    sortable_columns = sortable_columns(columns)
+    sort_by = all_params |> get_in_or_first("sort_by", sortable_columns) |> String.to_atom()
+    sort_dir = all_params |> get_in_or_first("sort_dir", @sort_dir) |> String.to_atom()
+    limit_options = Enum.map(limit_options, &to_string/1)
+    limit = all_params |> get_in_or_first("limit", limit_options) |> String.to_integer()
+    search = all_params["search"]
+    search = if search == "", do: nil, else: search
+    %{sort_by: sort_by, sort_dir: sort_dir, limit: limit, search: search}
+  end
+
+  defp sortable_columns(columns) do
+    Enum.flat_map(columns, &if(&1[:sortable], do: [to_string(&1[:field])], else: []))
+  end
+
+  defp get_in_or_first(params, key, valid) do
+    value = params[key]
+    if value in valid, do: value, else: hd(valid)
+  end
+
+  @impl true
+  def render(assigns) do
+    ~L"""
+    <div class="tabular-page">
+      <%= if @title do %>
```

Comment:
Will a title not be available at some point?

Reply:
Not anymore, it was part of a previous experiment :)